src/Xamarin.Android.Build.Tasks/Tasks/GenerateLibraryResources.cs

```diff
@@ -0,0 +1,200 @@
+﻿using Microsoft.Build.Framework;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Xamarin.Android.Tools;
+using Xamarin.Build;
+using ThreadingTasks = System.Threading.Tasks;
+
+namespace Xamarin.Android.Tasks
+{
+	/// <summary>
+	/// We used to invoke aapt/aapt2 per library (many times!), this task does the work to generate R.java for libraries without calling aapt/aapt2.
+	/// </summary>
+	public class GenerateLibraryResources : AsyncTask
+	{
+		/// <summary>
+		/// The main R.txt for the app
+		/// </summary>
+		[Required]
+		public string ResourceSymbolsTextFile { get; set; }
+
+		/// <summary>
+		/// The output directory for Java source code, such as: $(IntermediateOutputPath)android\src
+		/// </summary>
+		[Required]
+		public string OutputDirectory { get; set; }
+
+		/// <summary>
+		/// The list of R.txt files for each library
+		/// </summary>
+		public string [] LibraryTextFiles { get; set; }
+
+		/// <summary>
+		/// The accompanying manifest file for each library
+		/// </summary>
+		public string [] ManifestFiles { get; set; }
+
+		public override bool Execute ()
+		{
+			Yield ();
+			try {
+				var task = ThreadingTasks.Task.Run (() => {
+					DoExecute ();
+				}, CancellationToken);
+
+				task.ContinueWith (Complete);
+
+				base.Execute ();
+			} finally {
+				Reacquire ();
+			}
+
+			return !Log.HasLoggedErrors;
+		}
+
+		string output_directory;
+		Dictionary<Tuple<string, string>, string> main_r_txt;
+
+		void DoExecute ()
+		{
+			if (LibraryTextFiles == null || LibraryTextFiles.Length == 0)
+				return;
+
+			main_r_txt = new Dictionary<Tuple<string, string>, string> ();
+			using (var reader = File.OpenText (ResourceSymbolsTextFile)) {
+				foreach (var line in ParseFile (reader)) {
+					var key = new Tuple<string, string> (line [Index.Class], line [Index.Name]);
+					main_r_txt [key] = line [Index.Value];
+				}
+			}
+
+			Directory.CreateDirectory (OutputDirectory);
+			//So we don't need to consider WorkingDirectory on background threads
+			output_directory = Path.GetFullPath (OutputDirectory);
+
+			ThreadingTasks.ParallelOptions options = new ThreadingTasks.ParallelOptions {
+				CancellationToken = CancellationToken,
+				TaskScheduler = ThreadingTasks.TaskScheduler.Default,
+			};
+
+			ThreadingTasks.Parallel.For (0, LibraryTextFiles.Length, options, ProcessDirectory);
+		}
+
+		void ProcessDirectory (int index)
+		{
+			var r_txt = LibraryTextFiles [index];
+			if (!File.Exists (r_txt)) {
+				LogDebugMessage ($"Skipping, R.txt does not exist: {r_txt}");
```

Comment:
So I've found one case where this `R.txt` file doesn't exist. If we need to run the `<GetAdditionalResourcesFromAssemblies/>` task: 

![image](https://user-images.githubusercontent.com/840039/55258399-b8921500-5230-11e9-986d-31ed07c5e1c1.png)

It appears none of these zips we download contain `R.txt`:
```bash
$ for z in ~/.local/share/Xamarin/zips/*.zip; do zipinfo $z; done | grep R.txt
# no results
```
This kind of makes sense, because when I look in these zip files, the contents contain the _actual resource files_.

In the case where `R.txt` is missing, should we just write out the main app's `R.java` but with a different package name?

Reply:
That makes sense, that preserves the current behaviour.